lib/level10/games/game.ex

```diff
@@ -5,46 +5,196 @@ defmodule Level10.Games.Game do
   game's state, and then stored on the server to be updated or to serve data
   down to clients.
   """
-
   alias Level10.Games.{Card, Player}
 
+  @type join_code :: <<_::32>>
   @type t :: %__MODULE__{
-          complete: boolean(),
-          current_round: integer(),
-          discard_pile: list(Card.t()),
-          draw_pile: list(Card.t()),
-          hands: map(),
-          players: map(),
-          scoring: map(),
-          table: map()
+          current_round: non_neg_integer() | :pending | :completed,
+          discard_pile: [Card.t()],
+          draw_pile: [Card.t()],
+          hands: %{optional(Player.id()) => [Card.t()]},
+          join_code: join_code(),
+          players: [Player.t()],
+          scoring: %{optional(Player.id()) => {non_neg_integer(), non_neg_integer()}},
+          table: %{optional(Player.id()) => keyword()}
```

Comment:
If you wanted this to be even more strict it could be
```suggestion
          table: %{optional(Player.id()) => keyword(list(Card.t()))}
```